event/event_test.go

```diff
@@ -3,7 +3,6 @@ package event
 import (
 	"context"
 	"fmt"
-	"testing"
 )
 
 func Example() {
```

Comment:
@Arqu & @ramfox the example function shows how event subscription is re-worked to use a function argument instead of a return read-only channel. Consumers are welcome to subscribe to multiple event Topics in the same handler to keep the old multiplexing pattern: 

```go
var b event.Bus

func handler(ctx context.Context, topic event.Topic, payload interface{}) error {
  switch type {
    // place all your topic-oriented handlers here ...
    case event.ETDatasetPulled:
    case event.ETPeerConnected:
      // ...
  }
  return nil
}

b.Subscribe(handler, event.ETDatasetPulled, event.ETPeerConnected)
```

Reply:
Love it! Especially turning to sequential executiion first.

One thing I'd like to squeeze in here is the ability to subscribe to different "topic levels". Ie we have some topics like `main:fsi:linked`, `main:fsi:unlinked`. I'd like to be able to subscribe to either the specific one (`main:fsi:linked`) or subscribe to `main:fsi` and get the handler invoked on both events.

Happy to take this on, we can do a separate PR for it.

Reply:
I'd like to avoid implementing hierarchical events until we see a clear need for them, mainly out of performance concerns. 

Switching to synchronous events means the performance of event handlers affects the performance of qri generally. For that reason I'd like to err on the side of being _very_ careful about what events we incorporate, and how we expose those events.

I think we can get lots of the effects of "levels" by composing multiple event publishers where needed. If we wanted to test your example now, we could wire up an `fsi` event topic, and have the FSI subsystem fire them on all method calls. Put another way: I think we can have our cake & eat it too if we're careful about what event topics we create & fire

Reply:
Fair enough, this was mostly an ergonomic feature at this point with no clear need. Considering the single handler multi topic subscribe, I'm fairly good with it for the time being.